modules/mono/mono_gd/gd_mono_internals.cpp

```diff
@@ -139,4 +139,89 @@ void gd_unhandled_exception_event(MonoException *p_exc) {
 	args[0] = p_exc;
 	mono_runtime_invoke(unhandled_exception_method, nullptr, (void **)args, nullptr);
 }
+
+#if DEBUG_ENABLED
+static String _get_var_type(const Variant *p_var) {
+	String basestr;
+
+	if (p_var->get_type() == Variant::OBJECT) {
+		Object *bobj = *p_var;
+		if (!bobj) {
+			if (p_var->is_invalid_object()) {
+				basestr = "previously freed instance";
+			} else {
+				basestr = "null instance";
+			}
+		} else {
+			if (bobj->get_script_instance()) {
+				basestr = bobj->get_class() + " (" + bobj->get_script_instance()->get_script()->get_path().get_file() + ")";
+			} else {
+				basestr = bobj->get_class();
+			}
+		}
+
+	} else {
+		basestr = Variant::get_type_name(p_var->get_type());
+	}
+
+	return basestr;
+}
+
+static String _get_call_where(const String &p_method, const Variant *p_instance, const Variant **argptrs, int argc) {
```

Comment:
This logic was adjusted in the 4.x version, but I suppose we could also just keep this closer to the GDScript version here as it is basically entirely different code here anyways.

Reply:
I think it's safer to keep the code as close as the GDScript version for 3.x.